common/beerocks/bwl/dwpal/ap_wlan_hal_dwpal.cpp

```diff
@@ -1590,8 +1594,7 @@ bool ap_wlan_hal_dwpal::read_supported_channels()
 
 bool ap_wlan_hal_dwpal::set_tx_power_limit(int tx_pow_limit)
 {
-    LOG(DEBUG) << "set_tx_power_limit(): missing function implementation";
-    return true;
+    return m_nl80211_client->set_tx_power_limit(m_radio_info.iface_name, tx_pow_limit);
```

Comment:
Since now `nl80211_client` is passed to the constructor (and has a default value of `nullptr`), you can end up with a null reference exception here.
You should either remove the default value and make sure it's not `nullptr`, or check if it's not `nullptr` before every call.

Reply:
But in dwpal we always pass it, and override the default value - and if it fails then we assert with LOG_IF(FATAL) - do you still think it is necessary to check every call?

Reply:
No, I think you should remove the default value from the constructor and force passing a valid instance, while also validating that it's not nullptr.

Reply:
OK, done. I removed the default value and just moved the initialization to `base_wlan_hal_*`. IMO it makes much more sense there instead of manually specifying it for every derived class.